lib/libvarnish/vre.c

```diff
@@ -222,40 +240,66 @@ VRE_match(const vre_t *code, const char *subject, size_t length,
 	if (length == 0)
 		length = PCRE2_ZERO_TERMINATED;
 	vre_limit(code, lim);
-	return (vre_match(code, subject, length, 0, options, NULL));
+	return (vre_capture(code, subject, length, 0, options,
+	    NULL, NULL, NULL));
+}
+
+int
+VRE_capture(vre_t *code, const char *subject, size_t length, int options,
+    txt *groups, size_t count, const volatile struct vre_limits *lim)
+{
+	int i;
+
+	CHECK_OBJ_NOTNULL(code, VRE_MAGIC);
+	AN(subject);
+	AZ(options & (~VRE_MASK_MATCH));
```

Comment:
Not sure if we should really ban all options.

Reply:
You were on board when I added option smuggling prevention measures, to enforce what `vre.h` tries to convey: we define the options we want to support.

Otherwise #3657 gives you the escape hatch.

Reply:
I am on board, and I agree that `VRE_MASK_MATCH` can be adjusted.